### PR TITLE
[BugFix] Only remove partition vision infos when mv's partition has been dropped

### PR DESCRIPTION
**Pull request by LiShuMing**
Fixes #issue

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/main/java/com/starrocks/sql/common/SyncPartitionUtils.java [hunk 1]
```diff
@@ -612,8 +612,15 @@ private static void dropBaseVersionMetaForOlapTable(MaterializedView mv, String
                 mvToBaseMapping.values().forEach(parts -> parts.forEach(mvTableVersionMap::remove));
             }
         } else {
-            // This is a bad case for refreshing, and this problem will be optimized later.
-            versionMap.remove(tableId);
+            if (versionMap.containsKey(tableId)) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/common/SyncPartitionUtils.java:615]: same with    if (expr instanceof SlotRef) { branch

File: fe/fe-core/src/main/java/com/starrocks/sql/common/SyncPartitionUtils.java [hunk 2]
```diff
@@ -655,8 +662,17 @@ private static void dropBaseVersionMetaForExternalTable(MaterializedView mv, Str
             }
         } else {
             // This is a bad case for refreshing, and this problem will be optimized later.
-            versionMap.remove(new BaseTableInfo(tableName.getCatalog(), tableName.getDb(),
-                    baseTable.getName(), baseTable.getTableIdentifier()));
+            BaseTableInfo baseTableInfo = new BaseTableInfo(tableName.getCatalog(), tableName.getDb(),
+                    baseTable.getName(), baseTable.getTableIdentifier());
+            if (versionMap.containsKey(baseTableInfo)) {
+                Map<String, MaterializedView.BasePartitionInfo> baseTableVersionInfoMap = versionMap.get(baseTableInfo);
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/common/SyncPartitionUtils.java:668]: do need to check ICEBERG_ALL_PARTITION?

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 35 / 68 (51.47%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/common/SyncPartitionUtils.java|21|52|40.38%|[585, 586, 587, 588, 590, 591, 592, 594, 595, 596, 597, 598, 601, 603, 613, 615, 616, 623, 655, 656, 657, 658, 659, 660, 662, 663, 664, 668, 670, 674, 677]|
|:large_blue_circle:|com/starrocks/scheduler/PartitionBasedMvRefreshProcessor.java|11|13|84.62%|[457, 459]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 37 / 73 (50.68%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/common/SyncPartitionUtils.java|23|57|40.35%|[585, 586, 587, 588, 590, 591, 592, 594, 595, 596, 597, 598, 601, 603, 613, 615, 616, 623, 659, 660, 661, 662, 663, 664, 666, 667, 668, 669, 673, 676, 680, 682, 683, 686]|
|:large_blue_circle:|com/starrocks/scheduler/PartitionBasedMvRefreshProcessor.java|11|13|84.62%|[457, 459]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 46 / 87 (52.87%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/common/SyncPartitionUtils.java|24|62|38.71%|[586, 587, 588, 589, 591, 592, 593, 594, 595, 596, 597, 598, 600, 601, 604, 605, 614, 616, 617, 624, 625, 663, 664, 665, 666, 667, 668, 670, 671, 672, 673, 677, 680, 684, 687, 688, 689, 692]|
|:large_blue_circle:|com/starrocks/scheduler/PartitionBasedMvRefreshProcessor.java|19|22|86.36%|[499, 500, 501]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 52 / 94 (55.32%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/common/SyncPartitionUtils.java|24|62|38.71%|[586, 587, 588, 589, 591, 592, 593, 594, 595, 596, 597, 598, 600, 601, 604, 605, 614, 616, 617, 624, 625, 663, 664, 665, 666, 667, 668, 670, 671, 672, 673, 677, 680, 684, 687, 688, 689, 692]|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|2|3|66.67%|[750]|
|:large_blue_circle:|com/starrocks/scheduler/PartitionBasedMvRefreshProcessor.java|21|24|87.50%|[499, 500, 501]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|5|5|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 53 / 96 (55.21%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/common/SyncPartitionUtils.java|25|64|39.06%|[586, 587, 588, 589, 591, 592, 593, 594, 595, 596, 597, 598, 600, 601, 604, 605, 614, 616, 617, 624, 625, 664, 665, 666, 667, 668, 669, 671, 672, 673, 674, 675, 679, 682, 686, 689, 690, 691, 694]|
|:large_blue_circle:|com/starrocks/alter/AlterJobMgr.java|2|3|66.67%|[750]|
|:large_blue_circle:|com/starrocks/scheduler/PartitionBasedMvRefreshProcessor.java|21|24|87.50%|[499, 500, 501]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|5|5|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Pull request closed without merging**